frontend/packages/ceph-storage-plugin/src/components/create-storage-system/reducer.ts

```diff
@@ -188,6 +193,15 @@ export const reducer: WizardReducer = (prevState, action) => {
     case 'securityAndNetwork/setKms':
       newState.securityAndNetwork.kms = action.payload;
       break;
+    case 'securityAndNetwork/setVault':
+      newState.securityAndNetwork.kms.vault = action.payload;
+      break;
+    case 'securityAndNetwork/setIbmKms':
+      newState.securityAndNetwork.kms.ibmKms = action.payload;
+      break;
+    case 'securityAndNetwork/setKmsProvider':
+      newState.securityAndNetwork.kms.kmsProvider = action.payload;
+      break;
```

Comment:
Let's store the entire KMS object, we will try to ruse single key instead of two  IbmKms, vault

Reply:
Currently `securityAndNetwork/setKms` is used to updated entire KMS object (which contains state of all KMS providers, IBM/Vault etc, could be more in future). It is used for the case where we need to reset the entire KMS state (of all providers).

`securityAndNetwork/setIbmKms` & `securityAndNetwork/setVault` are KMS provider specific. No need to pass entire KMS object, only pass the value of the particular provider we want to update.

If I only use `securityAndNetwork/setKms` didn't I have to pass/update entire KMS object (containing states of all the providers) to the redux every time ?? This way I am only updating the state of particular provider.

Reply:
We are keeping 3 copies of the same data in different objects. We can use a switch case inside securityAndNetwork/setKms, so we can update specific provider kms info.

I am trying to group KMS keys as much as possible, so when we do any changes in the future it will be easy to make.

Reply:
using this method in common KMS related pages we dont need to use a lot of if conditions to use provider-specific key names like state.ibmKms (or) state.vault.

If you feel any issue with this, please comment here

Reply:
Got it, LGTM